### ISSUE TITLE
Deprecated memoryType in nvcc_detail/hip_runtime_api.h

### ISSUE THREAD

**Issue by AJcodes** [ROCm-Developer-Tools/HIP#2145]
When attempting to compile CUDA 11 with any of the newer ROCm versions (3.5 and above), the following error is returned
```
/opt/rocm/include/hip/nvcc_detail/hip_runtime_api.h(1355): error: class "cudaPointerAttributes" has no member "memoryType"
```

Based on some digging, it appears that `memoryType` was deprecated in CUDA 10, and has most likely been removed in CUDA 11

**Comment by doctorcolinsmith** [comment]
This is apparently fixed as of ROCm 3.8.

**Comment by SarbojitAMD** [comment]
Fixed and upstreamed hence closing the ticket
